packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -662,9 +662,13 @@ abstract class StatelessWidget extends Widget {
 
   /// Describes the part of the user interface represented by this widget.
   ///
-  /// The framework calls this method when this widget is inserted into the
-  /// tree in a given [BuildContext] and when the dependencies of this widget
-  /// change (e.g., an [InheritedWidget] referenced by this widget changes).
+  /// The framework calls this method when this widget is inserted into the tree
+  /// in a given [BuildContext] and when the dependencies of this widget change
+  /// (e.g., an [InheritedWidget] referenced by this widget changes). However,
+  /// these are not the only place this method gets called. This method can be
```

Comment:
Maybe say something around that it could potentially be called every frame and should not have any side effects.